pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -942,6 +950,30 @@ private static Set<String> getSegmentPartitionedColumns(@Nullable TableConfig ta
     return segmentPartitionedColumns;
   }
 
+  /**
+   * Retrieve multivalued columns for a table.
+   * From the table Schema , we get the multi valued columns of dimension fields.
+   *
+   * @param tableCache
+   * @param tableName
+   * @return multivalued columns of the table .
+   */
+  private static Set<String> getMultiValuedColumns(TableCache tableCache, String tableName) {
```

Comment:
This is not efficient. We can just look up on demand for each column in the aggregation

Reply:
But here we just get the Schema once and use it to do override in the entire query . If we want to do on-demand for each column when traversing the query , we need to loop the entire schema again to see if that's a multivalued column, which I think would be much more ineffective.  Other option I see is, holding the Set<> MultiValued columns in Schema class during its creation , to prevent looping the schema during each query.

Reply:
We still want to read schema once, but no need to collect the MV columns up front. You can use `schema.getFieldSpecFor(col)` to lookup the `FieldSpec` for a column. The assumption here is that we don't usually have that many aggregation functions comparing to number of columns